benchmark/solkz/compositional_field/CMakeLists.txt

```diff
@@ -33,7 +33,16 @@ ENDIF ()
 DEAL_II_INITIALIZE_CACHED_VARIABLES()
 
 SET(TARGET "solkz_compositional_field")
+SET(TARGET_DEPENDENCY "solkz")
+
 PROJECT(${TARGET})
 
+INCLUDE_DIRECTORIES("..")
+
 ADD_LIBRARY(${TARGET} SHARED solkz_compositional_field.cc)
-ASPECT_SETUP_PLUGIN(${TARGET})	
+ADD_LIBRARY(${TARGET_DEPENDENCY} SHARED ../solkz.cc)
```

Comment:
Ah, I think I understand now. You're trying to access that variable from cmake before it's actually set up (which happens further down in the `CMakeFiles.txt` file). But I do think that the variable is actually set inside the `.cc` file being compiled. What happens if you simply do
```
  #include ASPECT_SOURCE_DIR "/benchmarks/solkz/solkz.cc"
```
or the appropriate equivalent?